### PR TITLE
EASY-2307 duplicate license

### PR DESCRIPTION
**Pull request by jo-pol**
fixes EASY-2307 duplicate license

#### When applied it will
* produce the specified license when the EMD has another license entry than 'accept'
* 
* 

#### Where should the reviewer @DANS-KNAW/easy start?

@lindareijnhoudt
The new logic is in the methods `getSpecifiedLicense(...).getOrElse(toLicense(...)`
The rest is refactoring and additional tests.

#### How should this be manually tested?
* build and deploy on deasy
* [x] submit a dataset with the deposit-ui
* [x] on deasy
  ```
  easy-add-new-deposit-agreement -d easy-dataset:12 -o agreement.txt --doUpdate
  ```
* login as `archie001` on the webui, go to the administration tab of both datasets
  `https://deasy.dans.knaw.nl/ui/datasets/id/easy-dataset:12/tab/3`
* download the (last) agreement: 

#### related pull requests on github
repo                       | PR
-------------------------- | -----------------
easy-                      | [PR#](PRlink)

### CODE REVIEW

File: src/test/scala/nl/knaw/dans/easy/agreement/internal/PlaceholderMapperSpec.scala [hunk 1]
```diff
@@ -200,233 +206,258 @@ class PlaceholderMapperSpec extends UnitSpec with MockFactory with BeforeAndAfte
...
+    val items: Seq[MetadataItem] = Seq(
+      new BasicString("foo"),
+      new Relation("bar"),
+      new Relation("rabarbera", new URI("http://xx.dans.knaw.nl/yy")),
+    )
+    emd.getTerm _ expects term returning items.asJava
+
+    testInstance.metadataTable(emd, Seq("abc", "def"), "datasetID:1234").asScala should contain theSameElementsAs Seq(
+      Map(
+        MetadataKey.keyword -> null,
```
Comment (jo-pol) [inline src/test/scala/nl/knaw/dans/easy/agreement/internal/PlaceholderMapperSpec.scala]: * I supose this should be something like "relations"
* I this broken by my refactoring?

**Comment by rvanheest** [comment]
See https://github.com/DANS-KNAW-jp/easy-deposit-agreement-creator/pull/1 for fixing a TODO and some more review comments

File: src/main/scala/nl/knaw/dans/easy/agreement/internal/PlaceholderMapper.scala [hunk 2]
```diff
@@ -142,50 +142,74 @@ class PlaceholderMapper(metadataTermsFile: File)(implicit parameters: BaseParame
...
-            }.mkString(newLine)
-          case (t, items) => t.getName -> items.mkString(newLine)
-        }.reduce[(Term.Name, String)] { case ((t1, s1), (t2, s2)) if t1 == t2 => (t1, s1 + newLine * 2 + s2) }
+      .groupBy { case (term, _) => getDisplayName(term) }
+      .map { case (displayName, termsAndItems) =>
+        // by definition of `groupBy`, `termsAndItems` cannot be empty
+        val termName = termsAndItems.head._1.getName
+        val value = termsAndItems
+          .map((format _).tupled)
+          .reduce[String] {
```
Comment (rvanheest) [inline src/main/scala/nl/knaw/dans/easy/agreement/internal/PlaceholderMapper.scala]: let's make this `.reduce(_ + newLine * 2 + _)` instead.

Comment (rvanheest) [reply src/main/scala/nl/knaw/dans/easy/agreement/internal/PlaceholderMapper.scala]: I'll apply this change myself after merging...

**Pull request merged**